app/src/main/java/com/mikkelthygesen/android/tid_bma_java/CircularSlider.java

```diff
@@ -0,0 +1,306 @@
+        }
+        setThumbSize(thumbSize);
+        setThumbImage(thumbImage);
+        setThumbColor(thumbColor);
+
+        // assign padding - check for version because of RTL layout compatibility
+        int padding;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+            int all = getPaddingLeft() + getPaddingRight() + getPaddingBottom() + getPaddingTop() + getPaddingEnd() + getPaddingStart();
+            padding = all / 6;
+        } else {
+            padding = (getPaddingLeft() + getPaddingRight() + getPaddingBottom() + getPaddingTop()) / 4;
+        }
+        setPadding(padding);
+
+        a.recycle();
+    }
+
+    /* ***** Setters ***** */
+    public void setStartAngle(double startAngle) {
+        mStartAngle = startAngle;
+    }
+
+    public void setAngle(double angle) {
+        mAngle = angle;
+    }
+
+    public void setThumbSize(int thumbSize) {
+        mThumbSize = thumbSize;
+    }
+
+    public void setBorderThickness(int circleBorderThickness) {
+        mBorderThickness = circleBorderThickness;
+    }
+
+    public void setBorderColor(int color) {
+        mBorderColor = color;
+    }
+
+    public void setBorderGradientColors(String[] colors) {
+        mBorderGradientColors = new int[colors.length];
+        for (int i = 0; i < colors.length; i++) {
+            mBorderGradientColors[i] = Color.parseColor(colors[i]);
+        }
+    }
+
+    @SuppressWarnings("unused")
+    public void setBorderGradientColors(int[] colors) {
+        if (colors == null) {
+            mBorderGradientColors = null;
+            mGradientShader = null;
+        } else {
+            mBorderGradientColors = Arrays.copyOf(colors, colors.length);
+            mGradientShader = new SweepGradient(mCircleRadius, mCircleRadius, mBorderGradientColors, null);
+        }
+    }
+
+    public void setThumbImage(Drawable drawable) {
+        mThumbImage = drawable;
+    }
+
+    public void setThumbColor(int color) {
+        mThumbColor = color;
+    }
+
+    public void setPadding(int padding) {
+        mPadding = padding;
+    }
+
+    @Override
+    protected void onSizeChanged(int w, int h, int oldW, int oldH) {
+        // use smaller dimension for calculations (depends on parent size)
+        int smallerDim = w > h ? h : w;
+
+        // find circle's rectangle points
+        int largestCenteredSquareLeft = (w - smallerDim) / 2;
+        int largestCenteredSquareTop = (h - smallerDim) / 2;
+        int largestCenteredSquareRight = largestCenteredSquareLeft + smallerDim;
+        int largestCenteredSquareBottom = largestCenteredSquareTop + smallerDim;
+
+        // save circle coordinates and radius in fields
+        mCircleCenterX = largestCenteredSquareRight / 2 + (w - largestCenteredSquareRight) / 2;
+        mCircleCenterY = largestCenteredSquareBottom / 2 + (h - largestCenteredSquareBottom) / 2;
+        mCircleRadius = smallerDim / 2 - mBorderThickness / 2 - mPadding;
+
```

Comment:
it would be nice to have a rectangle data structure in a different class. then you can extract subfunctions for the square surrounding the circle